rideshare.rb

```diff
@@ -0,0 +1,132 @@
+ride_info = [
+  {
+    "DR0001": [
+      {
+        "date": "3rd Feb 2016",
+        "cost": 10,
+        "rider_ID": "RD0003",
+        "rating": 3 
+      },
+      {
+        "date": "3rd Feb 2016",
+        "cost": 30,
+        "rider_ID": "RD0015",
+        "rating": 4
+      },
+      {
+        "date": "5th Feb 2016",
+        "cost": 45,
+        "rider_ID": "RD0003",
+        "rating": 2
+      }
+    ],
+    "DR0002": [
+      {
+        "date": "3rd Feb 2016",
+        "cost": 25,
+        "rider_ID": "RD0073",
+        "rating": 5 
+      },
+      {
+        "date": "4th Feb 2016",
+        "cost": 15,
+        "rider_ID": "RD0013",
+        "rating": 1
+      },
+      {
+        "date": "5th Feb 2016",
+        "cost": 35,
+        "rider_ID": "RD0066",
+        "rating": 3
+      }  
+    ],
+    "DR0003": [
+      {
+        "date": "4th Feb 2016",
+        "cost": 5,
+        "rider_ID": "RD0066",
+        "rating": 5
+      },
+      {
+        "date": "5th Feb 2016",
+        "cost": 50,
+        "rider_ID": "RD0003",
+        "rating": 2
+      }
+    ],
+    "DR0004": [
+      {
+        "date": "3rd Feb 2016",
+        "cost": 5,
+        "rider_ID": "RD0022",
+        "rating": 5
+      },
+      {
+        "date": "4th Feb 2016",
+        "cost": 10,
+        "rider_ID": "RD0022",
+        "rating": 4
+      },
+      {
+        "date": "5th Feb 2016",
+        "cost": 20,
+        "rider_ID": "RD0073",
+        "rating": 5
+      }  
+    ]
+  }
+]
+
+# - the number of rides each driver has given
+ride_info.each do |driver|
+  driver.each do |id, rides|
+    puts "Driver #{id} has given #{rides.length} rides"
+  end
+end
+
+# - the total amount of money each driver has made
+ride_info.each do |driver|
+  driver.each do |id, rides|
+    total_made = rides.reduce(0) {|sum, ride| sum + ride[:cost]}
+    puts "Driver #{id} has made $#{total_made}"
+  end
+end
```

Comment:
You're looping over `ride_info` a few times, is there any way that you could loop over it less by utilizing methods?